sycl/source/detail/sampler_impl.cpp

```diff
@@ -59,6 +59,9 @@ RT::PiSampler sampler_impl::getOrCreateSampler(const context &Context) {
   PI_CALL((m_contextToSampler[Context] = RT::piSamplerCreate(
       getSyclObjImpl(Context)->getHandleRef(), sprops, &errcode_ret),
       errcode_ret));
+
+  if (errcode_ret == PI_INVALID_OPERATION)
```

Comment:
Suggest checking for image support like you do in accessors.

Reply:
According to OpenCL spec:
`clCreateSamplerWithProperties returns ... a NULL value with one of the following error values returned in errcode_ret: CL_INVALID_OPERATION if images are not supported by any device associated with context...`.
So I guess there's no need to create another `clGetDeviceInfo` call to get information that you will have after piSamplerCreate call.

Reply:
OK.